pandas/core/indexes/period.py

```diff
@@ -545,20 +550,12 @@ def get_loc(self, key, method=None, tolerance=None):
             key = Period(key, freq=self.freq)
         except ValueError:
             # we cannot construct the Period
-            raise KeyError(key)
+            raise KeyError(orig_key)
 
-        ordinal = key.ordinal if key is not NaT else key.value
         try:
-            return self._engine.get_loc(ordinal)
+            return Index.get_loc(self, key, method, tolerance)
```

Comment:
I wonder if this changes perf? the reason is that we have already created an _int64index, but now we are creating something new?

Reply:
> but now we are creating something new?

I don't think we're creating a new object here

Reply:
the engine itself is (maybe) being created again

Reply:
i guess on the first all, but on the flip side the _int64index may not have been created yert